woocommerce/woocommerce
# Order Object & API: Show 'line_discount'

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:
Save the price difference between the sales and list price in new meta `_line_discount`




Closes #30308



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. create a product with a regular and sales price
2. create a woocommerce order with that product
3. call the order rest api and check the new line item
4. or get the order object and check the new line item



### Changelog entry






<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type



-   [] Add - Adds functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>
Add a 'line_discount' line item for all orders which saves the difference between the regular price and the sales price for products that are discounted and ordered. Visible in the Order Object and Order API. 
 
#### Comment 

</details>
